File: Algorithms/Implementation Challenges
/Apple and Orange/AppleAndOrange.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

	public static boolean InHouse(int f_pos, int t_pos, int h_left, int h_right){
    	int p = t_pos + f_pos;
    	return (p >= h_left && p <= h_right);
	}
        
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int s = in.nextInt();
        int t = in.nextInt();
        int a = in.nextInt();
        int b = in.nextInt();
        int m = in.nextInt();
        int n = in.nextInt();
        int[] apple = new int[m];
        int n_a = 0, n_o = 0;
        for(int apple_i=0; apple_i < m; apple_i++){
            apple[apple_i] = in.nextInt();
            if(InHouse(apple[apple_i], a, s, t))	
				n_a++;

        }
        int[] orange = new int[n];
        for(int orange_i=0; orange_i < n; orange_i++){
            orange[orange_i] = in.nextInt();
            if(InHouse(orange[orange_i], b, s, t))	
				n_o++;
        }
        System.out.println(n_a);
        System.out.println(n_o);
    }
}

File: Algorithms/Implementation/Apple and Orange/AppleAndOrange.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {
	public static bool InHouse(int f_pos, int t_pos, int h_left, int h_right){
    	int p = t_pos + f_pos;
    	return (p >= h_left && p <= h_right);
	}

    static void Main(String[] args) {
        string[] tokens_s = Console.ReadLine().Split(' ');
        int s = Convert.ToInt32(tokens_s[0]);
        int t = Convert.ToInt32(tokens_s[1]);
        string[] tokens_a = Console.ReadLine().Split(' ');
        int a = Convert.ToInt32(tokens_a[0]);
        int b = Convert.ToInt32(tokens_a[1]);
        string[] tokens_m = Console.ReadLine().Split(' ');
        int m = Convert.ToInt32(tokens_m[0]);
        int n = Convert.ToInt32(tokens_m[1]);
        string[] apple_temp = Console.ReadLine().Split(' ');
        int[] apple = Array.ConvertAll(apple_temp,Int32.Parse);
        string[] orange_temp = Console.ReadLine().Split(' ');
        int[] orange = Array.ConvertAll(orange_temp,Int32.Parse);
        
        int n_a = 0, n_o = 0;
        foreach(var x in apple){
			if(InHouse(x, a, s, t))	
				n_a++;
        }
        
        foreach(var x in orange){
			if(InHouse(x, b, s, t))	
				n_o++;
        }
        
        Console.WriteLine(n_a);
        Console.WriteLine(n_o);
    }
}